src/commands/pull.js

```diff
@@ -0,0 +1,69 @@
+const chalk = require('chalk');
+const mkdirp = require('mkdirp');
+const fs = require('fs');
+const path = require('path');
+const tiged = require('tiged');
+const { ApifyCommand } = require('../lib/apify_command');
+const { getLoggedClientOrThrow } = require('../lib/utils');
+
+class PullCommand extends ApifyCommand {
+    async run() {
+        const { args } = this.parse(PullCommand);
+        const apifyClient = await getLoggedClientOrThrow();
+        const cwd = process.cwd();
+
+        const { actorId } = args;
+        const actor = await apifyClient.actor(actorId).get();
+        if (!actor) throw new Error(`Cannot find actor with ID '${actorId}' in your account.`);
+        const { name, versions } = actor;
+
+        // TODO: Check is sorted chronologically
+        const lastVersion = versions[versions.length - 1];
+
+        const dirpath = path.join(cwd, name);
+        mkdirp.sync(dirpath);
+
+        switch (lastVersion.sourceType) {
+            case 'TARBALL':
```

Comment:
How can this work? It's not downloading the tarball at all?

Reply:
Source files are created right after build - downloading them instead.

Reply:
That's not how it works - when you create an actor, it defaults to the `SOURCE_FILES` type, so the `sourceFiles` array is populated. But then when you switch to `TARBALL` source type, the tarball URL is stored to the `tarballUrl` property, but the `sourceFiles` array stays populated. So this works, but it downloads the wrong files.

On the worker, we also download the tarball, not the `sourceFiles` array if source type is `TARBALL`: https://github.com/apify/apify-worker/blob/develop/src/actor/act2_build_job.js#L362